kernel/pipes.c

```diff
@@ -220,13 +220,58 @@ static int pipe_walk_op(struct k_thread *thread, void *data)
 
 	walk_data->bytes_available += desc->bytes_to_xfer;
 
+	thread->no_wake_on_timeout = true;
+
 	if (walk_data->bytes_available >= walk_data->bytes_requested) {
 		return 1;
 	}
 
 	return 0;
 }
 
+/**
+ * @brief Callback routine used to clean up loose ends
+ *
+ * This callback routine is invoked to catch a couple of corner cases that
+ * arise from the use of the [no_wake_on_timeout] and [timeout_expired] fields.
+ *
+ * 1. In the event that the request could not be satisfied, the wait queue is
+ *    re-walked to undo the setting of the [no_wake_on_timeout] field. Note
+ *    if a thread's timeout had expired in this window, the timeout operation
+ *    must be completed.
+ * 2. The final thread in the waiter list had insufficient data copied to/from
+ *    it to wake it, and its timeout had expired--ensure it wakes.
+ *
+ * @return 1 to stop further walking; 0 to continue walking
```

Comment:
Is there any case where it is needed ? All calls to this function ignores its return.